### PR TITLE
Enable mini-badges in /shield/

### PR DESCRIPTION
**Pull request by alexfernandez**
Solves issue #25. Works fine including tests.

@tufosa: please review.

Added shields to package descriptions.

### CODE REVIEW

File: bin/app.js [hunk 1]
```diff
@@ -58,10 +52,12 @@ exports.stopServer = function(callback) {
...
-			return response.status(403).send({error: 'package ' + packageName + ' not found.'});
+	packages.find(packageName, function(error, result) {
+		if (error) {
+			return response.status(503).send({error: 'database not available'});
+		}
+		if (!result) {
+			return response.status(403).send({error: 'package ' + packageName + ' not found'});
 		}
 		badges.compileBadge(packageName, (result.quality * 100).toFixed(2), function (err, png) {
 			response.setHeader('Content-type', 'image/png');
```
Comment (tufosa) [inline bin/app.js]: you are not using err or png. Remove

Comment (tufosa) [inline bin/app.js]: replace "err" by "error". What happens if there is an error and png is null?

File: bin/app.js [hunk 2]
```diff
@@ -70,11 +66,29 @@ function serveBadge (request, response) {
...
+		{
+			return response.status(503).send({error: 'database not available'});
+		}
+		if (!result) {
+			return response.status(403).send({error: 'package ' + packageName + ' not found'});
+		}
+		var queryString = request.url.substringFrom('?');
+		var score = Math.round((result.quality || 0) * 100) / 100;
+		badges.retrieveShield(packageName, score, queryString, function (err, svg) {
+			response.setHeader('Content-type', 'image/svg+xml');
```
Comment (tufosa) [inline bin/app.js]: replace "err" by "error". What happens if there is an error and svg is null?

File: lib/badges.js [hunk 3]
```diff
@@ -44,7 +74,7 @@ exports.compileBadge = function (name, quality, callback) {
 	var canvas = new Canvas(256, 64);
 	var ctx = canvas.getContext('2d');
 
-	var img = new Image();
+	var img = new Canvas.Image();
```
Comment (tufosa) [inline lib/badges.js]: be a little bit more verbose in names. What is img? And ctx?

File: lib/db.js [hunk 4]
```diff
@@ -26,23 +27,17 @@ connect();
...
 			log.error('Could not connect to MongoDB: %s', error);
 		} else {
 			//close db before assigning a new one
-			log.debug('opening database: ' +  config.mongoConnection);
+			log.debug('opening database: ' +  connectionString);
 			if (db) {
 				db.close(true);
 			}
 			db = result;
+			console.log('db: %s', db);
```
Comment (tufosa) [inline lib/db.js]: remove. use log.info or log.debug instead

File: lib/db.js [hunk 5]
```diff
@@ -86,16 +82,19 @@ exports.reconnect = function(callback) {
...
-		db.close(true, function (error, result) {
-			isClosed = true;
-			callback (error, result);
-		});
+	if (!db)
+	{
+		return callback(null);
+	}
+	db.close(true, function (error, result) {
+		console.trace('closing');
```
Comment (tufosa) [inline lib/db.js]: remove. use log.trace instead

File: lib/packages.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+	db.addCallback(function(error, result)
+	{
+		if (error)
+		{
+			return callback('Could not access database: ' + error);
+		}
+		packagesCollection = result.collection(config.packagesCollection);
+		if (packagesCollection)
+		{
+			console.log('was there');
```
Comment (tufosa) [inline lib/packages.js]: remove. use log.info or log.debug instead

File: lib/packages.js [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+			return callback(null, packagesCollection);
+		}
+			console.log('creating');
```
Comment (tufosa) [inline lib/packages.js]: remove. use log.info or log.debug instead

File: lib/packages.js [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+exports.close = function(callback)
+{
+	db.close(callback);
+};
+
+function testClose(callback)
+{
+	console.log('closing');
```
Comment (tufosa) [inline lib/packages.js]: remove. use log.info or log.debug instead

File: package.json [hunk 9]
```diff
@@ -21,6 +21,7 @@
 		"express": "*",
 		"grunt": "*",
 		"canvas": "*",
+		"basic-request": "*",
```
Comment (tufosa) [inline package.json]: We should unify our "requests" libs (we have request and basic-request). I am happy to use "basic-request" if it's simpler.

File: lib/db.js [hunk 10]
```diff
@@ -66,6 +61,7 @@ function connect() {
 exports.addCallback = function(callback) {
 	if (db) {
 		if (isClosed === true) {
+			console.log('closed; connecting');
```
Comment (tufosa) [inline lib/db.js]: remove. use log.info or log.debug instead

File: lib/packages.js [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+exports.update = function(object, callback)
+{
+	openCollection(function(error, collection)
+	{
+		if (error)
+		{
+			log.error('Could not open collection: %s', error);
+			return callback('database not available');
+		}
+		console.log('collection: %s', collection);
```
Comment (tufosa) [inline lib/packages.js]: remove. use log.info or log.debug instead

File: bin/app.js (hunk 1 above)
Comment (alexfernandez) [inline bin/app.js]: Solved in commit abed65a.

File: bin/app.js (hunk 2 above)
Comment (alexfernandez) [inline bin/app.js]: Solved in commit abed65a.

File: lib/badges.js (hunk 3 above)
Comment (alexfernandez) [inline lib/badges.js]: Solved in commit eb80c20.

File: lib/db.js (hunk 4 above)
Comment (alexfernandez) [inline lib/db.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: lib/db.js (hunk 5 above)
Comment (alexfernandez) [inline lib/db.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: lib/packages.js (hunk 6 above)
Comment (alexfernandez) [inline lib/packages.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: lib/packages.js (hunk 7 above)
Comment (alexfernandez) [inline lib/packages.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: lib/packages.js (hunk 8 above)
Comment (alexfernandez) [inline lib/packages.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: package.json (hunk 9 above)
Comment (alexfernandez) [inline package.json]: It should be simpler, but I will leave the changes to you if you don't mind.

File: lib/db.js (hunk 10 above)
Comment (alexfernandez) [inline lib/db.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

File: lib/packages.js (hunk 11 above)
Comment (alexfernandez) [inline lib/packages.js]: I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.

**Pull request merged**